bpf/lib/nodeport.h

```diff
@@ -1403,10 +1465,35 @@ static __always_inline int dsr_set_opt4(struct __ctx_buff *ctx,
 
 	return 0;
 }
+#elif DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+static __always_inline int dsr_encap_geneve_opt4(struct __ctx_buff *ctx,
+						 struct iphdr *ip4, __be32 svc_addr,
+						 __be16 svc_port, int *ifindex)
+{
+	struct remote_endpoint_info *info = NULL;
+	struct geneve_opt4 gopt;
+
+	info = ipcache_lookup4(&IPCACHE_MAP, ip4->daddr, V4_CACHE_KEY_LEN, 0);
+	if (info && info->tunnel_endpoint != 0) {
+		set_geneve_opt4(svc_port, svc_addr, &gopt);
+
+		return  __encap_with_nodeid_opt4(ctx, info->tunnel_endpoint,
+						 WORLD_ID,
```

Comment:
I overlooked this earlier - can we keep the TCP optimization? Sending the whole connection over the tunnel, but only adding the GENEVE option to the TCP-SYN.

Reply:
Sure, I will do that